### ISSUE TITLE
planning graph error for specific domain

### ISSUE THREAD

**Issue by trycharder** [JuliaPlanners/SymbolicPlanners.jl#18]
When running a domain and problem where the solution is to call one action before another, an error is raised when building the planning graph. It could certainly be the case that the domain or problem might be invalid semantically (both are included after the REPL output).

REPL Output + Stacktrace:
```jl
julia> VERSION
v"1.9.3"

(Test) pkg> status SymbolicPlanners
Project Test v0.0.1-alpha
Status `~/git/test/Project.toml`
  [b6fe5377] SymbolicPlanners v0.1.19

julia> using PDDL, SymbolicPlanners

julia> domain = load_domain("domain.pddl")
GenericDomain
  name: test
  predicates: Dict{Symbol, PDDL.Signature} with 3 entries
    :finish            => (finish)
    :start             => (start)
    Symbol("one-done") => (one-done)
  actions: 
    :one => (one)
    :two => (two)

julia> prob = load_problem("problem.pddl")
GenericProblem
  name: test0
  init: 1-element Vector{Term}
    (start)
  goal: (and (finish))

julia> state = initstate(domain, prob)
GenericState
  fluents: 1 (boolean)
    (start) => true

julia> planner = AStarPlanner(HAdd())
ForwardPlanner{Nothing}(HAdd(sum, precomputed=false), nothing, 1.0f0, 1.0f0, 9223372036854775807, Inf, false, false, false, nothing)

julia> spec = MinStepsGoal(prob)
MinStepsGoal(Term[finish])

julia> sol = planner(domain, state, spec)
ERROR: BoundsError: attempt to access 0-element Vector{Vector{Int64}} at index [1]
Stacktrace:
 [1] getindex
   @ ./essentials.jl:13 [inlined]
 [2] build_planning_graph(domain::GenericDomain, state::GenericState, goal::Compound; statics::Vector{Symbol}, relevants::Vector{Symbol})
   @ SymbolicPlanners ~/.julia/packages/SymbolicPlanners/CxHor/src/heuristics/pgraph.jl:124
 [3] #build_planning_graph#79
   @ ~/.julia/packages/SymbolicPlanners/CxHor/src/heuristics/pgraph.jl:28 [inlined]
 [4] build_planning_graph
   @ ~/.julia/packages/SymbolicPlanners/CxHor/src/heuristics/pgraph.jl:23 [inlined]
 [5] precompute!(h::HAdd, domain::GenericDomain, state::GenericState, spec::MinStepsGoal)
   @ SymbolicPlanners ~/.julia/packages/SymbolicPlanners/CxHor/src/heuristics/hsp.jl:79
 [6] solve(planner::ForwardPlanner{Nothing}, domain::GenericDomain, state::GenericState, spec::MinStepsGoal)
   @ SymbolicPlanners ~/.julia/packages/SymbolicPlanners/CxHor/src/planners/forward.jl:167
 [7] (::ForwardPlanner{Nothing})(domain::GenericDomain, state::GenericState, spec::MinStepsGoal)
   @ SymbolicPlanners ~/.julia/packages/SymbolicPlanners/CxHor/src/planners/planners.jl:18
 [8] top-level scope
   @ REPL[7]:1
```

domain.pddl
```pddl
(define (domain test)
 (:requirements :strips)
 (:predicates (finish) (start) (one-done))
 (:action one
  :parameters ()
  :precondition (and (start))
  :effect (and (one-done)))
 (:action two 
  :parameters ()
  :precondition (and (one-done))
  :effect (and (finish)))
)
```

problem.pddl
```pddl
(define (problem test0)
	(:domain test)
	(:objects)
	(:init
            (start)
	)
	(:goal (and (finish)
	))
)
```

Expected Solution
```
[one, two]
```

**Comment by trycharder** [comment]
If this is a valid bug, then it might originate on line 86 in `src/heuristics/pgraph.jl`. When modifying the line from:
```jl
 preconds = isempty(act.preconds) ? Term[Const(true)] : act.preconds
 ```
to:
```jl
 preconds = act.preconds
 ```
It does seem to fix the error, generate the expected output (for the examples above) and pass the unit tests, but I'm not sure but about any other downstream consequences of modifying that line.

**Comment by ztangent** [comment]
Thanks for flagging this! Looks like I wasn't handling actions with no preconditions correctly -- fixed by fedb439ba07eca853541332a84d3757e2036f001 and should be released soon.
